src/pages/settings/AboutPage/ConsolePage.tsx

```diff
@@ -0,0 +1,201 @@
+import {FlatList, View} from 'react-native';
+import type {ListRenderItem, ListRenderItemInfo} from 'react-native';
+import {withOnyx} from 'react-native-onyx';
+import type {OnyxEntry} from 'react-native-onyx';
+import Button from '@components/Button';
+import ConfirmModal from '@components/ConfirmModal';
+import HeaderWithBackButton from '@components/HeaderWithBackButton';
+import * as Expensicons from '@components/Icon/Expensicons';
+import ScreenWrapper from '@components/ScreenWrapper';
+import Text from '@components/Text';
+import TextInput from '@components/TextInput';
+import useKeyboardShortcut from '@hooks/useKeyboardShortcut';
+import useLocalize from '@hooks/useLocalize';
+import useThemeStyles from '@hooks/useThemeStyles';
+import {addLog} from '@libs/actions/Console';
+import {createLog, sanitizeConsoleInput} from '@libs/Console';
+import type {Log} from '@libs/Console';
+import localFileCreate from '@libs/localFileCreate';
+import localFileDownload from '@libs/localFileDownload';
+import Navigation from '@libs/Navigation/Navigation';
+import CONST from '@src/CONST';
+import ONYXKEYS from '@src/ONYXKEYS';
+import ROUTES from '@src/ROUTES';
+
+type CapturedLogs = Record<number, Log>;
+
+type ConsolePageOnyxProps = {
+    /** Logs captured on the current device */
+    capturedLogs: OnyxEntry<CapturedLogs>;
+
+    /** Whether or not logs should be stored */
+    shouldStoreLogs: OnyxEntry<boolean>;
+};
+
+type ConsolePageProps = ConsolePageOnyxProps;
+
+/**
+ * Loops through all the logs and parses the message if it's a stringified JSON
+ * @param logs Logs captured on the current device
+ * @returns CapturedLogs with parsed messages
+ */
+const parseStringifyMessages = (logs: Log[]) => {
+    if (isEmpty(logs)) {
+        return;
+    }
+
+    return logs.map((log) => {
+        try {
+            const parsedMessage = JSON.parse(log.message);
+            return {
+                ...log,
+                message: parsedMessage,
+            };
+        } catch {
+            // If the message can't be parsed, just return the original log
+            return log;
+        }
+    });
+};
+
+function ConsolePage({capturedLogs, shouldStoreLogs}: ConsolePageProps) {
+    const [input, setInput] = useState('');
+    const [logs, setLogs] = useState(capturedLogs);
+    const [isGeneratingLogsFile, setIsGeneratingLogsFile] = useState(false);
+    const [isLimitModalVisible, setIsLimitModalVisible] = useState(false);
+    const {translate} = useLocalize();
+    const styles = useThemeStyles();
+
+    const logsList = useMemo(() => (logs ? Object.values(logs).reverse() : []), [logs]);
+
+    useEffect(() => {
+        if (!shouldStoreLogs) {
+            return;
+        }
+
+        setLogs((prevLogs) => ({...prevLogs, ...capturedLogs}));
+    }, [capturedLogs, shouldStoreLogs]);
+
+    const executeArbitraryCode = () => {
+        const sanitizedInput = sanitizeConsoleInput(input);
+
```

Comment:
NAB: Extra line, we can avoid the new lines if the variable is used immediately in the next line.